src/helm/proxy/clients/auto_client.py

```diff
@@ -125,28 +128,30 @@ def make_request(self, request: Request) -> RequestResult:
         def make_request_with_retry(client: Client, request: Request) -> RequestResult:
             return client.make_request(request)
 
-        organization: str = request.model_organization
-        client: Client = self.get_client(request)
+        client: Client = self._get_client(request.model)
 
         try:
             return make_request_with_retry(client=client, request=request)
         except RetryError as e:
             last_attempt: Attempt = e.last_attempt
             retry_error: str = (
-                f"Failed to make request to {organization} after retrying {last_attempt.attempt_number} times"
+                f"Failed to make request to {request.model} after retrying {last_attempt.attempt_number} times"
             )
             hlog(retry_error)
 
             # Notify our user that we failed to make the request even after retrying.
             return replace(last_attempt.value, error=f"{retry_error}. Error: {last_attempt.value.error}")
 
-    def get_tokenizer_client(self, organization: str) -> Client:
-        """Return a client based on `organization`, creating it if necessary."""
-        client: Optional[Client] = self.tokenizer_clients.get(organization)
+    def _get_tokenizer_client(self, tokenizer: str) -> Client:
+        """Return a client based on the tokenizer, creating it if necessary."""
+        organization: str = tokenizer.split("/")[0]
+        client: Optional[Client] = self.tokenizer_clients.get(tokenizer)
 
         if client is None:
             cache_config: CacheConfig = self._build_cache_config(organization)
-            if organization in [
```

Comment:
for later: this if/else chain is kind of gross and seems like it would be better to factor out this functionality somehow to localize the changes when adding models

Reply:
This entire client instantiation branching logic needs to be rewritten at some point...